src/Microsoft.AspNetCore.Razor.Language/Extensions/FunctionsDirectivePass.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Extensions
+{
+    public class FunctionsDirectivePass : RazorIRPassBase, IRazorDirectiveClassifierPass
```

Comment:
> We should pick an approach of static XYZDirective class with a static register method and a nested private class (that's what InjectDirective does as an example) or what you have in this PR, static class, register, the directive class and a public pass.

Meh, every case is case-by-case.

In these cases, it might be useful to reuse the directive definition, but implement it another way. So you'd want the directive descriptor but not the pass, which means the pass has to be public so you can remove it.